infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -85,8 +79,8 @@ public DataStore(Map<String, Object> opts) {
      * obtain an in-memory cache, no content loader, no writer, no size limit.
      * @return  new LoadingCache for the DataStore
      */
-    private LoadingCache<String, String> buildMemoryDs() {
-        LoadingCache<String, String> cache = Caffeine
+    private Cache<String, Object> buildMemoryDs() {
+        Cache<String, Object> cache = Caffeine
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "cache" which hides the field declared at line 55. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "cache". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)